dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -787,25 +814,36 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
         Optional<SourceFile> mainDescriptor = filteredSourcefiles.stream().filter((sourceFile -> Objects.equals(sourceFile.getPath(), primaryDescriptorFilePath))).findFirst();
 
         boolean isValid = true;
+        boolean safe = false;
         StringBuilder validationMessage = new StringBuilder();
         Map<String, String> validationMessageObject = new HashMap<>();
 
         if (mainDescriptor.isPresent()) {
-            Yaml yaml = new Yaml();
-            String content = mainDescriptor.get().getContent();
-            if (content == null || content.isEmpty()) {
-                isValid = false;
-                validationMessage.append("Primary descriptor is empty.");
-            } else if (!content.contains("class: Workflow")) {
+            try {
+                Yaml safeYaml = new Yaml(new SafeConstructor());
+                // This should throw an exception if there are unexpected blocks
+                safeYaml.load(mainDescriptor.get().getContent());
+                safe = true;
+            } catch (Exception e) {
```

Comment:
Log the exception.

Perhaps add a validation message? On the one hand, how helpful do we want to be to people trying to attack exploit us? On the other hand, maybe the user is doing it inadvertently (they were in turn hacked), so...